src/main/java/streams/cta/stereo/Stereo.java

```diff
@@ -0,0 +1,296 @@
+    }
+
+
+    /**
+     * This class describes a plane in [x, y, z] space for a single telescope shower.
+     * The plane is aligned with the reconstructed angle of the shower in the camera.
+     * One plane for each camera partaking in an event is created. Using these planes both impact position
+     * and direction can be reconstructed.
+     *
+     */
+    static class Plane {
+        //the tlescope id this reconstructed plane belongs to
+        final int telescopeId;
+        //the weight given to the plane
+        final double weight;
+        //two vectors describing the plane
+        final double[] v1;
+        final double[] v2;
+
+        //the normalvector of that plane
+        final double[] normalVector;
+
+        //the position of the telescope on the ground
+        final double[] telescopePosition;
+
+        Plane(int id, double phi, double theta,  Data data) {
+            this.telescopeId = id;
+            double length = (double) data.get("telescope:" + id + ":shower:length");
+            double width = (double) data.get("telescope:" + id + ":shower:width");
+            double cogX = (double) data.get("telescope:" + id + ":shower:cog:x");
+            double cogY = (double) data.get("telescope:" + id + ":shower:cog:y");
+            double psi = (double) data.get("telescope:" + id + ":shower:psi");
+            double size = (double) data.get("telescope:" + id + ":shower:size");
+
+            TelescopeDefinition tel = CameraMapping.getInstance().telescopeFromId(id);
+
+            //get two points on the shower axis
+            double pX = cogX + length * cos(psi);
+            double pY = cogY + length * sin(psi);
+
+            double[] pDirection = cameraCoordinateToDirectionVector(pX, pY, phi, theta, tel.opticalFocalLength, 0);
+            double[] cogDirection = cameraCoordinateToDirectionVector(cogX, cogY, phi, theta, tel.opticalFocalLength, 0);
+
+            this.weight = size * (length / width);
+            this.v1 = cogDirection;
+            this.v2 = pDirection;
+
+            // c  = (v1 X v2) X v1
+            Vector3D c = Vector3D.crossProduct(Vector3D.crossProduct(new Vector3D(v1), new Vector3D(v2)), new Vector3D(v1));
+            Vector3D norm = Vector3D.crossProduct(new Vector3D(v1), c);
+
+            this.normalVector = norm.normalize().toArray();
+
+            telescopePosition = new double[]{tel.telescopePositionX, tel.telescopePositionY, tel.telescopePositionZ};
+        }
+
+        /**
+         * A small convenience function to return the [x, y, z] normal vector as an actual Vector3d object
+         *
+         * @return a the normal vector of the plane
+         */
+        Vector3D getNormalAsVector(){
+            return new Vector3D(normalVector);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
```

Comment:
personally, I don't like to inline the if-conditions. But this is just a my opinion.